### ISSUE TITLE
Catching AWS parameters that need Fn::Sub

### ISSUE THREAD

**Issue by snowman2** [awslabs/cfn-python-lint#588]
*cfn-lint version: 0.10.1

*Description of issue.*

When there is a string that has an `AWS::` parameter, it is not caught in the linting process.

For example, this will currently pass when it should fail:
```
 "Resource":  [
    "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:my_function"
 ],
```
Because it should be:

```
 "Resource":  [
    { "Fn:Sub": "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:my_function" }
 ],
```
```

**Comment by fatbasstard** [comment]
Hi @snowman2 ,

Can give give a bitter template as an example? If I fill in this code I get a `E1029 Found an embedded parameter outside of an "Fn::Sub"` error, as expected. So there might be some more context needed.

Do be aware that this cannot be raised as an error on all locations, [some properties are excluded](https://github.com/awslabs/cfn-python-lint/blob/master/src/cfnlint/rules/functions/SubNeeded.py#L29). You can read up the reasoning behind this is in the issue that triggered the creation of this rule: https://github.com/awslabs/cfn-python-lint/issues/186

**Comment by snowman2** [comment]
Ah, the resource properties are excluded. Good to know. Thanks!

**Comment by kddejong** [comment]
Yea sorry @snowman2 There were a few properties (usually via feedback) that we have found that are  hard to detect this on.  Its mostly cause there could be similar looking variables.   Think `${parameter}` or `${resource.attribute}`.  That being said we could have maybe taken a different approach to this solution.  The other would have been to build a list of valid list of ${...} items and look for those.  Its not like there could have been a lot more valid reasons for `${AWS::Region}`

**Comment by snowman2** [comment]
I think looking for the `AWS::` parameters would definitely be helpful if it is not conflicting.
